config.ru

```diff
@@ -11,6 +11,19 @@ map '/lobster' do
   run Rack::Lobster.new
 end
 
+map '/headers' do
+  headers = proc do |env|
+    [200, { "Content-Type" => "text/html" }, [
+      env.select {|key,val| key.start_with? 'HTTP_'}
+      .collect {|key, val| [key.sub(/^HTTP_/, ''), val]}
+      .collect {|key, val| CGI::escapeHTML "#{key}: #{val}"}
+      .sort
+      .join("<br/>\n")
+    ]]
+  end
+  run headers
```

Comment:
Could you specify more precisely what should `properly formatted HTML` mean? With `html` and `head` tags? Or something more? Alternatively probably this can serve `text/plain` instead of "text/html". WDYT?